### ISSUE TITLE
Keys

### ISSUE THREAD

**Issue by jbucaran** [picodom/picodom#9]
The current implementation of picodom **supports** [keyed elements](https://github.com/hyperapp/hyperapp/blob/master/docs/keys.md#keys).

This was an important feature in HyperApp, but is it just as important in picodom and can we avoid it? 

/cc @zaceno

**Comment by zaceno** [comment]
Yes, I think keys are equally as (if not more!) important in picodom as in hyperapp.

It's a feature that needs to be present in the vdom layer of any vdom-based web stack (be that hyperapp or picodom+whatever), in order to enable all the use-cases we talked about when we introduced keys in hyperapp.

I don't think we can avoid keys if we want picodom to be generally useful.

There may be ways to make it more abstract, so as to allow userland to turn keys on or off or use some alternate pattern to achieve the same goals (akin to the idea of introducing snabbdom style hooks in lieu of the current lifecycle methods) But ultimately that would just make picodom *more* complicated -- not less. I don't yet see the need for that :)
